product_combination_exclude/models/product_product.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Graeme Gellatly
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+
+    _inherit = 'product.product'
+
+    @api.multi
+    def create(self, vals):
+        vals = dict(vals or {})
```

Comment:
Odoo idiom, its everywhere in their latest code.  I prefer it to the if x then y else z